pattern/constraint/thing/ValueConstraint.java

```diff
@@ -317,7 +317,7 @@ public void addTo(Traversal traversal) {
 
     public static class Variable extends ValueConstraint<ThingVariable> {
 
-        public Variable(ThingVariable owner, GraqlToken.Predicate.Equality predicate, ThingVariable variable) {
+        public Variable(ThingVariable owner, GraqlToken.Predicate predicate, ThingVariable variable) {
```

Comment:
This change is not good - now it makes it look like we can do a say `substring` or `like` (regex) comparision between two variables, which we don't actually support! I'd revert this and undo moving the big `if` switch back into its prior place